src/py21cmfast/drivers/coeval.py

```diff
@@ -0,0 +1,901 @@
+        pointer_fields = []
+        for cls in [InitialConditions, PerturbedField, IonizedBox, BrightnessTemp]:
+            pointer_fields += cls.get_pointer_fields()
+
+        if spin_temp:
+            pointer_fields += TsBox.get_pointer_fields()
+
+        if hbox:
+            pointer_fields += HaloBox.get_pointer_fields()
+
+        return pointer_fields
+
+    @property
+    def user_params(self):
+        """User params shared by all datasets."""
+        return self.brightness_temp_struct.user_params
+
+    @property
+    def cosmo_params(self):
+        """Cosmo params shared by all datasets."""
+        return self.brightness_temp_struct.cosmo_params
+
+    @property
+    def flag_options(self):
+        """Flag Options shared by all datasets."""
+        return self.brightness_temp_struct.flag_options
+
+    @property
+    def astro_params(self):
+        """Astro params shared by all datasets."""
+        return self.brightness_temp_struct.astro_params
+
+    @property
+    def random_seed(self):
+        """Random seed shared by all datasets."""
+        return self.brightness_temp_struct.random_seed
+
+    def _particular_rep(self):
+        return ""
+
+    def _write_particulars(self, fname):
+        for name in ["init", "perturb", "ionization", "brightness_temp", "spin_temp"]:
+            struct = getattr(self, f"{name}_struct")
+            if struct is not None:
+                struct.write(fname=fname, write_inputs=False)
+
+                # Also write any other inputs to any of the constituent boxes
+                # to the overarching attrs.
+                with h5py.File(fname, "a") as fl:
+                    for inp in struct._inputs:
+                        if inp not in fl.attrs and inp not in [
+                            "user_params",
+                            "cosmo_params",
+                            "flag_options",
+                            "astro_params",
+                            "global_params",
+                        ]:
+                            fl.attrs[inp] = getattr(struct, inp)
+
+    @classmethod
+    def _read_particular(cls, fname):
+        kwargs = {}
+
+        with h5py.File(fname, "r") as fl:
+            for output_class in _ut.OutputStruct._implementations():
+                if output_class.__name__ in fl:
+                    kwargs[_ut.camel_to_snake(output_class.__name__)] = (
+                        output_class.from_file(fname)
+                    )
+
+        return kwargs
+
+    def __eq__(self, other):
+        """Determine if this is equal to another object."""
+        return (
+            isinstance(other, self.__class__)
+            and other.redshift == self.redshift
+            and self.user_params == other.user_params
+            and self.cosmo_params == other.cosmo_params
+            and self.flag_options == other.flag_options
+            and self.astro_params == other.astro_params
+        )
+
```

Comment:
This is a much older issue but we don't have random seed in `__eq__`?

Reply:
That's true. We should probably ensure that as well -- problematically sometimes you also want to compare without the random seed. But I think you're right that the _default_ equality method should include the random seed.